pkg/internal/limitrange/transformer.go

```diff
@@ -45,98 +46,78 @@ func NewTransformer(ctx context.Context, namespace string, lister corev1listers.
 
 		// The assumption here is that the min, max, default, ratio have already been
 		// computed if there is multiple LimitRange to satisfy the most (if we can).
-		// Count the number of containers (that we know) in the Pod.
+		// Count the number of "step" containers in the Pod.
 		// This should help us find the smallest request to apply to containers
-		nbContainers := len(p.Spec.Containers)
-		// FIXME(#4230) maxLimitRequestRatio to support later
-		defaultLimits := getDefaultLimits(limitRange)
-		defaultInitRequests := getDefaultInitContainerRequest(limitRange)
-		for i := range p.Spec.InitContainers {
-			// We are trying to set the smallest requests possible
-			if p.Spec.InitContainers[i].Resources.Requests == nil {
-				p.Spec.InitContainers[i].Resources.Requests = defaultInitRequests
-			} else {
-				for _, name := range resourceNames {
-					setRequestsOrLimits(name, p.Spec.InitContainers[i].Resources.Requests, defaultInitRequests)
-				}
-			}
-			// We are trying to set the highest limits possible
-			if p.Spec.InitContainers[i].Resources.Limits == nil {
-				p.Spec.InitContainers[i].Resources.Limits = defaultLimits
-			} else {
-				for _, name := range resourceNames {
-					setRequestsOrLimits(name, p.Spec.InitContainers[i].Resources.Limits, defaultLimits)
-				}
+		stepContainers := []*corev1.Container{}
+		for i, c := range p.Spec.Containers {
+			if strings.HasPrefix(c.Name, "step-") {
+				stepContainers = append(stepContainers, &p.Spec.Containers[i])
 			}
 		}
+		// No step containers, bail early we don't have anything to transform.
+		if len(stepContainers) == 0 {
+			return p, nil
+		}
 
-		defaultRequests := getDefaultAppContainerRequest(limitRange, nbContainers)
-		for i := range p.Spec.Containers {
-			if p.Spec.Containers[i].Resources.Requests == nil {
-				p.Spec.Containers[i].Resources.Requests = defaultRequests
-			} else {
-				for _, name := range resourceNames {
-					setRequestsOrLimits(name, p.Spec.Containers[i].Resources.Requests, defaultRequests)
-				}
-			}
-			if p.Spec.Containers[i].Resources.Limits == nil {
-				p.Spec.Containers[i].Resources.Limits = defaultLimits
+		defaultStepRequests := getDefaultStepContainerRequest(limitRange, len(stepContainers))
+
+		// Empty LimitRange Requests, bail early we don't have anything to transform.
+		if len(defaultStepRequests) == 0 {
+			return p, nil
+		}
+
+		for _, c := range stepContainers {
+			if c.Resources.Requests == nil {
+				c.Resources.Requests = defaultStepRequests
 			} else {
 				for _, name := range resourceNames {
-					setRequestsOrLimits(name, p.Spec.Containers[i].Resources.Limits, defaultLimits)
+					setRequests(name, c.Resources.Requests, defaultStepRequests)
```

Comment:
See comment above.

Reply:
Are you sure? I think this will work when the step container doesn't have any limits, but if the step or sidecar container does have limits, I think the pod we end up creating could be rejected by the limitrange